tests/PHPStan/Analyser/data/bcmath-dynamic-return-php8.php

```diff
@@ -0,0 +1,97 @@
+// Returns the result as a numeric-string, or FALSE if modulus is 0 or exponent is negative.
+\PHPStan\Testing\assertType('false', bcpowmod('10', '-2', '0')); // exponent negative, and modulus is 0
+\PHPStan\Testing\assertType('false', bcpowmod('10', '-2', '1')); // exponent negative
+\PHPStan\Testing\assertType('false', bcpowmod('10', '2', $nonNumeric)); // Warning: bcmath function argument is not well-formed
+\PHPStan\Testing\assertType('false', bcpowmod('10', '-2', '-1')); // exponent negative
+\PHPStan\Testing\assertType('false', bcpowmod('10', '-2', -1.3)); // exponent negative
+\PHPStan\Testing\assertType('false', bcpowmod('10', -$iPos, '-1')); // exponent negative
+\PHPStan\Testing\assertType('false', bcpowmod('10', -$iPos, '1')); // exponent negative
+\PHPStan\Testing\assertType('false', bcpowmod('10', $nonNumeric, $nonNumeric)); // Warning: bcmath function argument is not well-formed
+\PHPStan\Testing\assertType('false', bcpowmod($iPos, $nonNumeric, $nonNumeric));
+\PHPStan\Testing\assertType('false', bcpowmod('10', '2', '0')); // modulus is 0
+\PHPStan\Testing\assertType('false', bcpowmod('10', 2.3, '0')); // modulus is 0
+\PHPStan\Testing\assertType('false', bcpowmod('10', '0', '0')); // modulus is 0
+\PHPStan\Testing\assertType('numeric-string', bcpowmod('10', '0', '-2'));
+\PHPStan\Testing\assertType('numeric-string', bcpowmod('10', '2', '2'));
+\PHPStan\Testing\assertType('numeric-string', bcpowmod('10', $iUnknown, '2'));
+\PHPStan\Testing\assertType('numeric-string', bcpowmod($iPos, '2', '2'));
+\PHPStan\Testing\assertType('numeric-string|false', bcpowmod('10', $mixed, $mixed));
+\PHPStan\Testing\assertType('numeric-string', bcpowmod('10', '2', '2'));
+\PHPStan\Testing\assertType('numeric-string', bcpowmod('10', -$iNeg, '2'));
+\PHPStan\Testing\assertType('numeric-string', bcpowmod('10', $nonNumeric, '2')); // Warning: bcmath function argument is not well-formed
+\PHPStan\Testing\assertType('numeric-string|false', bcpowmod('10', $iUnknown, $iUnknown));
+
+//  bcsqrt ( string $operand [, int $scale = 0 ] ) : string
+// Returns the square root as a numeric-string, or NULL if operand is negative.
+\PHPStan\Testing\assertType('numeric-string', bcsqrt('10', $iNeg));
+\PHPStan\Testing\assertType('numeric-string', bcsqrt('10', 1));
+\PHPStan\Testing\assertType('numeric-string', bcsqrt('0.00', 1));
+\PHPStan\Testing\assertType('numeric-string', bcsqrt(0.0, 1));
+\PHPStan\Testing\assertType('numeric-string', bcsqrt('0', 1));
+\PHPStan\Testing\assertType('numeric-string|null', bcsqrt($iUnknown, $iUnknown));
+\PHPStan\Testing\assertType('numeric-string', bcsqrt('10', $iPos));
+\PHPStan\Testing\assertType('null', bcsqrt('-10', 0)); // Warning: Square root of negative number
+\PHPStan\Testing\assertType('null', bcsqrt($iNeg, 0));
+\PHPStan\Testing\assertType('null', bcsqrt('10', $nonNumeric)); // Warning: Second argument must be ?int (Fatal in PHP8)
```

Comment:
when it fatals in php8, it should return `*NEVER*`

Reply:
This PR was focused on `bcdiv()` and `bcmod()` only.

I just added fixes for `bcpowmod()` and `bcsqrt()`.